tools/clang/test/CodeGenHLSL/quick-test/mad_opt.hlsl

```diff
@@ -0,0 +1,9 @@
+// RUN: %dxc -E main -T ps_6_0 %s | FileCheck %s
```

Comment:
Need more tests:

1. UMad/IMad
2. Check for constants in different args (as suggested below)
3. Make sure we don't transform floats unless fast math is enabled